q2_assembly/helpers/helpers.py

```diff
@@ -116,15 +117,35 @@ def collate_alignments(alignments: BAMDirFmt) -> BAMDirFmt:
 
 
 def collate_genomes(
-    genomes: Union[DNAFASTAFormat, GenomeSequencesDirectoryFormat]
+    genomes: Union[DNAFASTAFormat, GenomeSequencesDirectoryFormat],
+    on_duplicates: str = "error",
 ) -> GenomeSequencesDirectoryFormat:
     genomes_dir = GenomeSequencesDirectoryFormat()
     if isinstance(genomes[0], DNAFASTAFormat):
+        error_on_duplicates = True if on_duplicates == "error" else False
+        has_duplicates = False
+        ids = set()
+        duplicate_ids = set()
         for genome_file in genomes:
             for genome in genome_file.view(DNAIterator):
                 name = genome.metadata["id"]
                 with open(os.path.join(genomes_dir.path, name + ".fasta"), "w") as f:
                     skbio.io.write(genome, format="fasta", into=f)
+                if name in ids:
+                    has_duplicates = True
```

Comment:
Since you keep a separate list of duplicated IDs yo do not need this variable. You can just check later `if duplicate_ids`.